libbeat/generator/fields/fields.go

```diff
@@ -153,11 +172,40 @@ func writeIndentedLine(buf *bytes.Buffer, line string, indent int) error {
 }
 
 // Generate collects fields.yml files and concatenates them into one global file.
-func Generate(esBeatsPath, beatPath string, files []*YmlFile, output string) error {
-	files, err := collectCommonFiles(esBeatsPath, beatPath, files)
+func Generate(esBeatsPath, beatPath string, moduleFiles []*YmlFile, output string) error {
+	libbeatFiles, err := collectCommonFiles(esBeatsPath, beatPath)
 	if err != nil {
 		return err
 	}
 
+	beatFiles, err := collectBeatFields(beatPath)
+	if err != nil {
+		return err
+	}
+
+	if err = os.MkdirAll("build", 0755); err != nil {
+		return err
+	}
+
+	// Write separate files into build directory for use by fields.go generation
+	err = writeGeneratedFieldsYml(libbeatFiles, filepath.Join(beatPath, "build/libbeat.yml"))
```

Comment:
I started putting all these files into `build/fields/`. Can you drop them in there instead of build/.

https://github.com/elastic/beats/pull/9842/files#diff-408f5df10fc1cfd92fc66134de4d8edbR28